### PR TITLE
Add Save button to Document editor

### PR DESCRIPTION
**Pull request by luorlandini**
- Add Save button to Document editor 

- Editors save button sends the resource metadata edited inside the detail panel to the backend

https://user-images.githubusercontent.com/4085786/123756971-c5918900-d8bd-11eb-80de-11fd76f2f44b.mov

### CODE REVIEW

File: geonode_mapstore_client/client/js/epics/gnsave.js [hunk 1]
```diff
@@ -128,22 +129,33 @@ const SaveAPI = {
...
+
     }
 };
 
 export const gnSaveContent = (action$, store) =>
     action$.ofType(SAVE_CONTENT)
         .switchMap((action) => {
             const state = store.getState();
-            const contentType = state.gnresource?.type || 'map';
+            const contentType = state.gnresource?.type || state.gnresource?.data.resource_type || 'map';
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/epics/gnsave.js:148]: `state.gnresource?.type` is a property we set client side once the viewer open (eg: https://github.com/GeoNode/geonode-mapstore-client/blob/6b874a2305cc86735dfbc6e291c0490ff19055b4/geonode_mapstore_client/client/js/epics/gnviewer.js#L149).
 I think we should do the same for document config here
https://github.com/GeoNode/geonode-mapstore-client/blob/6b874a2305cc86735dfbc6e291c0490ff19055b4/geonode_mapstore_client/client/js/epics/gnviewer.js#L187-L202
 and do not rely on the resource_type from api

File: geonode_mapstore_client/client/js/epics/gnsave.js [hunk 2]
```diff
@@ -167,14 +179,16 @@ export const gnSaveDirectContent = (action$, store) =>
...
-            const resourceId = mapInfo?.id // injected map id
+            const resourceId = mapInfo?.id
             || state?.gnresource?.id; // injected geostory id
             return Observable.defer(() => getResourceByPk(resourceId))
                 .switchMap((resource) => {
                     const metadata = {
-                        name: resource?.title,
-                        description: resource?.abstract,
-                        thumbnail: resource?.thumbnail_url
+                        name: (state.gnresource.data.name !== resource?.title ) ? state.gnresource.data.name : resource?.title,
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/epics/gnsave.js:187]: we colud use the selector or a constant instaed to reapeat state.gnresource.data to ensure state.gnresource.data exist

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: geonode_mapstore_client/client/js/epics/gnsave.js (hunk 1 above)
Comment (luorlandini) [reply geonode_mapstore_client/client/js/epics/gnsave.js]: ok, I implement the setResourceType

File: geonode_mapstore_client/client/js/epics/gnsave.js (hunk 2 above)
Comment (luorlandini) [reply geonode_mapstore_client/client/js/epics/gnsave.js]: ok, implement selectors

**Review by allyoucanmap — APPROVED** [review]

**Pull request merged**